lightning/src/ln/outbound_payment.rs

```diff
@@ -1031,7 +1042,7 @@ impl OutboundPayments {
 					let payment_hash = PaymentHash(Sha256::hash(&keysend_preimage.0).to_byte_array());
 					let pay_params = PaymentParameters::from_static_invoice(invoice);
 					let mut route_params = RouteParameters::from_payment_params_and_value(pay_params, amount_msat);
-					route_params.max_total_routing_fee_msat = *max_total_routing_fee_msat;
+					route_params.max_total_routing_fee_msat = route_params_config.map(|params| params.max_total_routing_fee_msat).flatten();
```

Comment:
This doesn't compile since `route_param_config` is not an option. FYI, use the following to make sure the compiler sees this code: `!RUSTFLAGS="--cfg=async_payments" cargo +1.63.0 check -p lightning`.

Reply:
Thanks a lot for the suggestion on the code! It will be super helpful for future references too!

I also noticed that the function was not using the rest of params_config when building PaymentParams. I have also fixed this bug in [pr3342.06](https://github.com/shaavan/rust-lightning/commits/pr3342.06)

Thanks for the pointer!